webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/RemoteSourceDescriptorService.scala

```diff
@@ -59,63 +62,43 @@ class RemoteSourceDescriptorService @Inject()(dSRemoteWebknossosClient: DSRemote
-    val uri = new URI(pathLiteral)
-    if (DataVaultService.isRemoteScheme(uri.getScheme)) {
-      uri
-    } else if (uri.getScheme == null || uri.getScheme == DataVaultService.schemeFile) {
-      val localPath = Path.of(uri.getPath)
-      if (localPath.isAbsolute) {
-        if (localPath.toString.startsWith(localDatasetDir.getParent.toAbsolutePath.toString) || dataStoreConfig.Datastore.localDirectoryWhitelist
-              .exists(whitelistEntry => localPath.toString.startsWith(whitelistEntry)))
-          uri
-        else
-          throw new Exception(
-            s"Absolute path $localPath in local file system is not in path whitelist. Consider adding it to datastore.localDirectoryWhitelist")
-      } else { // relative local path, resolve in dataset dir
-        val pathRelativeToDataset = localDatasetDir.resolve(localPath).normalize
-        val pathRelativeToLayer = localDatasetDir.resolve(layerName).resolve(localPath).normalize
-        if (pathRelativeToDataset.toFile.exists) {
-          pathRelativeToDataset.toUri
-        } else {
-          pathRelativeToLayer.toUri
-        }
-      }
-    } else {
-      throw new Exception(s"Unsupported path: $localDatasetDir")
-    }
-  }
-
-  def resolveMagPath(datasetDir: Path, layerDir: Path, layerName: String, magLocator: MagLocator): URI =
+  def resolveMagPath(localDatasetDir: Path, layerDir: Path, layerName: String, magLocator: MagLocator): UPath =
     magLocator.path match {
       case Some(magLocatorPath) =>
-        uriFromPathLiteral(magLocatorPath, datasetDir, layerName)
+        if (magLocatorPath.isAbsolute) {
+          magLocatorPath
+        } else {
+          // relative local path, resolve in dataset dir
+          val pathRelativeToDataset = localDatasetDir.resolve(magLocatorPath.toLocalPathUnsafe).normalize
+          val pathRelativeToLayer =
+            localDatasetDir.resolve(layerName).resolve(magLocatorPath.toLocalPathUnsafe).normalize
+          if (pathRelativeToDataset.toFile.exists) {
+            UPath.fromLocalPath(pathRelativeToDataset)
+          } else {
+            UPath.fromLocalPath(pathRelativeToLayer)
+          }
+        }
       case _ =>
         val localDirWithScalarMag = layerDir.resolve(magLocator.mag.toMagLiteral(allowScalar = true))
         val localDirWithVec3Mag = layerDir.resolve(magLocator.mag.toMagLiteral())
         if (localDirWithScalarMag.toFile.exists) {
-          localDirWithScalarMag.toUri
+          UPath.fromLocalPath(localDirWithScalarMag)
         } else {
-          localDirWithVec3Mag.toUri
+          UPath.fromLocalPath(localDirWithVec3Mag)
         }
     }
 
-  private def uriForMagLocator(baseDir: Path,
-                               dataSourceId: DataSourceId,
-                               layerName: String,
-                               magLocator: MagLocator): Box[URI] = tryo {
+  private def resolveMagPath(baseDir: Path,
```

Comment:
During first reading it wasn't so easy to see what the difference between this `resolveMagPath` and the one above is. Or more like, why are both needed. Maybe things might be a little clearer if one of the two methods gets a small renaming?

But not so super important imo